rita_client/src/operator_update/tests.rs

```diff
@@ -0,0 +1,248 @@
+
+        let result = parse_keys(key_file);
+        for item in result {
+            assert_eq!(item, removed_keys[0].to_string());
+        }
+
+        remove_temp_file(key_file).unwrap();
+    }
+    #[test]
+    fn test_authorized_keys_create_if_missing() {
+        let added_keys = vec![
+            String::from("ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHFgFrnSm9MFS1zpHHvwtfLohjqtsK13NyL41g/zyIhK test@hawk-net ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDVF1POOko4/fTE/SowsURSmd+kAUFDX6VPNqICJjn8eQk8FZ15WsZKfBdrGXLhl2+pxM66VWMUVRQOq84iSRVSVPA3abz0H7JYIGzO8psTweSZfK1jwHfKDGQA1h1aPuspnPrX7dyS1qLZf3YeVUUi+BFsW2gSiMadbS4zal2c2F1AG5Ezr3zcRVA8y3D0bZxScPAEX74AeTFcimHpHFyzDtUsRpf0uSEXZcMFqX5j4ETKlIs28k1v8LlhHo91IQYHEtbyi/I1M0axbF4VCz5JlcbAs9LUEJg8Kx8LxzJSeSJbxVwyk5WiEDwVsCL2MAtaOcJ+/FhxLb0ZEELAHnXFNSqmY8QoHeSdHrGP7FmVCBjRb/AhVUHYvsG94rO3Ij4H5XsbsQbP3AHVKbvf387WB53Wga7VrBXvRC9aDisetdP9+4/seVIBbOIePotaiHoTyS1cJ+Jg0PkKy96enqwMt9T1Wt8jURB+s/A/bDGHkjB3dxomuGxux8dD6UNX54M= test-rsa@hawk-net"),
+        ];
+        let removed_keys: Vec<String> = vec![];
+        let key_file: &str = "create_keys_file";
+        let _update = update_authorized_keys(added_keys, removed_keys, key_file);
+        assert!(Path::new(key_file).exists());
+    }
+    #[test]
+    fn test_usage_data_processing() {
+        // this tests the flow used in rita client's operator update loop used to process usage data sent up to ops
+        let dummy_usage_tracker = generate_dummy_usage_tracker_temp();
+        let mut usage_data_client = dummy_usage_tracker.client_bandwidth.clone();
+        // Test the sort function first:
+        // shuffle the data because it's currently ordered
+        usage_data_client
+            .make_contiguous()
+            .shuffle(&mut rand::thread_rng());
+        println!(
+            "Sample of current shuffle is {} {} {}",
+            usage_data_client.get(0).unwrap().index,
+            usage_data_client.get(1).unwrap().index,
+            usage_data_client.get(2).unwrap().index
+        );
+        // Sort by index increasing so newest data at back. Note that usage hours are stored to disk as
+        // the opposite order where newest are added to the front, so this is inefficient.
+        // Options here to optimize are either a/write my own binary sort again which will compare for the existing structure
+        // where the saved vecdeque is highest index to lowest index, b/rework usage tracker so that we save data lowest index
+        // to highest index, or c/the current solution(inefficient, as we will be fully reversing the whole vecdeque of each
+        // client and relay every checkin): sort the entire list in reverse order to use with the builtin bin search from vecdeque
+
+        // this here sorts from lowest index to highest index, so we end with a vecdeque that we can use binary search on
+        usage_data_client
+            .make_contiguous()
+            .sort_by(|a, b| a.index.cmp(&b.index));
+        println!(
```

Comment:
you are testing what is currently the same code but might change.. break out whats in the update loop into a function so you can call that function here.